actgen/agents/directed_qnet.py

```diff
@@ -40,24 +43,57 @@ def directed_update(self, states, actions, delta_update, n_updates, baseline_qne
                 if self.optim == 'adam':
                     optimizer = torch.optim.Adam(list(self.parameters()), lr=self.lr)
 
-                with torch.no_grad():
-                    original_q_values = self.forward(s.float())
-                    q_target = original_q_values.clone()
-                    q_target[0][a] = float(original_q_values[0][a]) + delta_update
+                # update for normal DQN
+                if self.agent_type == 'dqn':
```

Comment:
Seems like there's a fair bit of duplication here. Can we consolidate the dqn and action_dqn parts at all?

Reply:
I agree, I've also thought about this. 

I don't know if we can consolidate anything in this script, because 1) the shape of tensors 2) which ones are the input, are really different. Maybe we can use a `DQN` class that inherits `MLP`, and some of the code can go there. But this is essentially just moving the duplicate code into another file, and doesn't really solve the problem.